### PR TITLE
[stdlib] Improve large integer support

### PR DESCRIPTION
**Pull request by natecook1000**
This adds support for converting `DoubleWidth` integers with values outside the range of `Int64`/`UInt64` to strings, and for initializing `DoubleWidth` instances with literals that are outside that range as well.

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please benchmark

**Comment by natecook1000** [comment]
@swift-ci Please test

File: stdlib/public/core/StringLegacy.swift [hunk 1]
```diff
@@ -213,6 +213,65 @@ extension String {
...
+  let radix = T.Magnitude(radix)
+  let quotientAndRemainder: (T.Magnitude) -> (T.Magnitude, T.Magnitude) =
+    radixIsPowerOfTwo
+      ? { ( $0 &>> radix.trailingZeroBitCount, $0 & (radix - 1) ) }
+      : { $0.quotientAndRemainder(dividingBy: radix) }
+
+  let isNegative = T.isSigned && value < 0
+  var value = value.magnitude
+
+  var result: [UInt8] = []
```
Comment (moiseev) [inline stdlib/public/core/StringLegacy.swift]: Maybe worth calling `reseveCapacity` here? It's a really a nano-optimization, but may make a difference even if the capacity is, say, 4.

File: test/stdlib/DoubleWidth.swift [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+  _ = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0
+    as UInt256
+}
+
+dwTests.test("Arithmetic/unsigned") {
+  let x: DoubleWidth<UInt8> = 1000
+  let y: DoubleWidth<UInt8> = 1111
+  expectEqual(x + 1, 1001)
+  expectEqual(x + x, 2000)
+  expectEqual(x - (1 as DoubleWidth<UInt8>), 999)
```
Comment (moiseev) [inline test/stdlib/DoubleWidth.swift]: Is this `as` needed? Is there still a `Strideable` overload for `-` kicking int?

**Review by moiseev — COMMENTED** [review]
LGTM

File: test/stdlib/DoubleWidth.swift (hunk 2 above)
Comment (natecook1000) [reply test/stdlib/DoubleWidth.swift]: Yep: `test/stdlib/DoubleWidth.swift:74:17: error: ambiguous use of operator '-'` along with a pile of warnings.

<details>
<summary>A pile of warnings</summary>
<pre><code>test/stdlib/DoubleWidth.swift:75:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:76:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(y - x, 111)
                ^
test/stdlib/DoubleWidth.swift:72:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:75:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:76:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(y - x, 111)
                ^
test/stdlib/DoubleWidth.swift:93:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:95:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - (1 as DoubleWidth<Int8>), 999)
                ^
test/stdlib/DoubleWidth.swift:96:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:97:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(0 - x, -1000)
                ^
test/stdlib/DoubleWidth.swift:99:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - y, 2111)
                ^
test/stdlib/DoubleWidth.swift:93:17: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x + 1, 1001)
                ^
test/stdlib/DoubleWidth.swift:95:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - (1 as DoubleWidth<Int8>), 999)
                ^
test/stdlib/DoubleWidth.swift:96:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - x, 0)
                ^
test/stdlib/DoubleWidth.swift:97:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(0 - x, -1000)
                ^
test/stdlib/DoubleWidth.swift:99:17: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectEqual(x - y, 2111)
                ^
test/stdlib/DoubleWidth.swift:166:38: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.min) - 1))
                                     ^
test/stdlib/DoubleWidth.swift:167:38: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.max) + 1))
                                     ^
test/stdlib/DoubleWidth.swift:169:26: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.min) - 1 < Int.min)
                         ^
test/stdlib/DoubleWidth.swift:170:26: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.max) + 1 > Int.max)
                         ^
test/stdlib/DoubleWidth.swift:166:38: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.min) - 1))
                                     ^
test/stdlib/DoubleWidth.swift:167:38: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectNil(Int(exactly: DW(Int.max) + 1))
                                     ^
test/stdlib/DoubleWidth.swift:169:26: warning: '-' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.min) - 1 < Int.min)
                         ^
test/stdlib/DoubleWidth.swift:170:26: warning: '+' is deprecated: Please use explicit type conversions or Strideable methods for mixed-type arithmetics.
  expectTrue(DW(Int.max) + 1 > Int.max)
</code></pre>
</details>

File: stdlib/public/core/StringLegacy.swift (hunk 1 above)
Comment (natecook1000) [reply stdlib/public/core/StringLegacy.swift]: After testing for a bit, I can't find an amount to reserve that actually makes a difference… Our "amortized O(1)" complexity for `append(_:)` is pretty real!

**Pull request merged**